### PR TITLE
PS-8993 Convert audit_log_ filter plugin into component_audit_log_filter component (8.1)

### PR DESCRIPTION
**Pull request by alina-derkach-oaza**
new file:   docs/audit-log-filter-compression-encryption.md
	new file:   docs/audit-log-filter-formats.md
	new file:   docs/audit-log-filter-json.md
	new file:   docs/audit-log-filter-naming.md
	new file:   docs/audit-log-filter-new.md
	new file:   docs/audit-log-filter-old.md
	new file:   docs/audit-log-filter-overview.md
	new file:   docs/audit-log-filter-restrictions.md
	new file:   docs/audit-log-filter-security.md
	new file:   docs/audit-log-filter-variables.md
	new file:   docs/disable-audit-log-filter.md
	new file:   docs/filter-audit-log-filter-files.md
	new file:   docs/install-audit-log-filter.md
	new file:   docs/manage-audit-log-filter.md
	new file:   docs/reading-audit-log-filter-files.md
	new file:   docs/uninstall-audit-log-filter.md
	modified:   mkdocs-base.yml

### CODE REVIEW

File: docs/audit-log-filter-overview.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Audit Log Filter overview
+
+The Audit Log Filter component allows you to monitor, log, and block a connection or query actively executed on the selected server. 
+
+Enabling the component produces a log file that contains a record of server activity. The log file has information on connections and databases accessed by that connection. 
+
+The component uses the `mysql` system database to store filter and user account data. Set the [`audit_log_filter_database`](audit-log-filter-variables.md#audit_log_filter_database) variable at server startup to select a different database.
```
Comment (oleksandr-kachan) [inline docs/audit-log-filter-overview.md]: audit_log_filter_database -> audit_log_filter.database

File: docs/audit-log-filter-overview.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+This privilege is defined by the server and enables the user to configure the component.
+
+### `AUDIT_ABORT_EXEMPT`
+
+This privilege allows queries from a user account to always be executed. An `abort` item does not block them. This ability lets the user account regain access to a system if an audit is misconfigured. The query is logged due to the privilege. User accounts with the `SYSTEM_USER` privilege have the `AUDIT_ABORT_EXEMPT` privilege.
+
+## Audit Log Filter tables
+
+The Audit Log Filter component uses `mysql` system database tables in the `InnoDB` storage engine. These tables store user account data and filter data. When you start the server, change the component's database with the `audit_log_filter_database` variable.
```
Comment (oleksandr-kachan) [inline docs/audit-log-filter-overview.md]: audit_log_filter_database -> audit_log_filter.database

File: docs/audit-log-filter-overview.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+The `component_audit_log_filter` table stores the definitions of the filters and has the following column definitions:
```
Comment (oleksandr-kachan) [inline docs/audit-log-filter-overview.md]: Tables name stays unchanged component_audit_log_filter -> audit_log_filter

File: docs/audit-log-filter-variables.md [hunk 4]
```diff
@@ -0,0 +1,728 @@
...
+The following audit log filter functions are available.
+
+| Function name |
+| --- | 
+| [audit_log_encryption_password_get(keyring_id)](#audit_log_encryption_password_getkeyring_id) |
+| [audit_log_encryption_password_set(new_password)](#audit_log_encryption_password_setnew_password) |
+| [audit_log_filter_flush()](#audit_log_filter_flush) |
+| [audit_log_read()](#audit_log_read) |
+| [audit_log_read_bookmark()](#audit_log_read_bookmark) |
+| [audit_log_session_filter_id()](#audit_log_session_filter_id() |
```
Comment (oleksandr-kachan) [inline docs/audit-log-filter-variables.md]: Not sure but looks like link format is broken here

File: docs/audit-log-filter-variables.md [hunk 5]
```diff
@@ -0,0 +1,728 @@
...
+
+This function returns the internal ID of the audit log filter in the current session.
+
+Returns 0 (zero) if the session has no assigned filter.
+
+### audit_log_filter_remove_filter(filter_name)
+
+This function removes the selected filter from the current set of filters.
+
+If user accounts are assigned the selected filter, the user accounts are no longer filtered. The user accounts are removed from `audit_log_filter_user`. If the user accounts are in a current session, they are detached from the selected filter and no longer logged.
```
Comment (oleksandr-kachan) [inline docs/audit-log-filter-variables.md]: audit_log_filter_user -> audit_log_user

File: docs/audit-log-filter-variables.md [hunk 6]
```diff
@@ -0,0 +1,728 @@
...
+    | audit_log_filter_set_user('user-name@localhost', 'filter-name')  |
+    +-------------------------------------------------------------------+
+    | OK                                                                |
+    +-------------------------------------------------------------------+
+    ```
+
+## Audit log filter options and variables
+
+| Name |
+| --- |
```
Comment (oleksandr-kachan) [inline docs/audit-log-filter-variables.md:341]: I've probably forgot to describe this properly but all sys vars names format changed to audit_log_filter.[some_name]. Each name has 'audit_log_filter' prefix followed by a '.' and actual name.

File: docs/audit-log-filter-variables.md [hunk 7]
```diff
@@ -0,0 +1,728 @@
...
+| [`audit-log-filter`](#audit-logvalue) |
+| [`audit_log_buffer_size`](#audit_log_buffer_size) |
```
Comment (oleksandr-kachan) [inline docs/audit-log-filter-variables.md]: audit_log_filter.buffer_size

File: docs/audit-log-filter-variables.md [hunk 8]
```diff
@@ -0,0 +1,728 @@
...
+| [`audit_log_filter_compression`](#audit_log_filter_filter_compression) |
```
Comment (oleksandr-kachan) [inline docs/audit-log-filter-variables.md]: audit_log_filter.compression and similar for all the following names

File: docs/audit-log-filter-variables.md [hunk 9]
```diff
@@ -0,0 +1,728 @@
...
+* OFF
+* FORCE
+* FORCE_PLUS_PERMANENT
+
+
+### `audit_log_filter_buffer_size`
+
+| Option name | Description |
+| --- | --- |
+| Command-line | --audit-log-filter-buffer-size |
```
Comment (oleksandr-kachan) [inline docs/audit-log-filter-variables.md]: Names in this format should be changed as well --audit-log-filter-buffer-size -> --audit-log-filter.buffer-size

File: docs/disable-audit-log-filter.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
+# Disable Audit Log Filter logging
+
+The `audit_log_filter_disable` system variable lets you disable or enable logging for all connections.
```
Comment (oleksandr-kachan) [inline docs/disable-audit-log-filter.md]: audit_log_filter.disable

File: docs/disable-audit-log-filter.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+Setting audit_log_filter_disable has the following effect:
+
+| Value | Actions |
+|---|---|
+| true | Generates a warning. Audit log function calls and changes in variables generate session warnings. Disables the component. |
+| false | Re-enables the component and generates a warning. This is the default value. |
+
+## Privileges required
+
+Setting the value of `audit_log_filter_disable` at runtime requires the following:
```
Comment (oleksandr-kachan) [inline docs/disable-audit-log-filter.md]: audit_log_filter.disable

File: docs/filter-audit-log-filter-files.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+|---|---|---|
+| audit_log_filter_flush() | Manually flush the filter tables | `SELECT audit_log_filter_flush()`
+| audit_log_filter_set_filter() | Defines a filter | `SELECT audit_log_filter_set_filter('log_connections','{ "filter":{}}'`')
+| audit_log_filter_remove_filter() | Removes a filter |
+| audit_log_filter_set_user() | Assigns a filter to a specific user account |
+| audit_log_filter_remove_user() | Removes the filters from a specific user account |
+
+Using a SQL interface, you can define, display, or modify audit log filters. The filters are stored in the `mysql` system database. 
+
+A read-only variable, `audit_log_filter_id`, signals if a filter is assigned to a specific session.
```
Comment (oleksandr-kachan) [inline docs/filter-audit-log-filter-files.md]: audit_log_filter_id replaced with audit_log_session_filter_id() function

File: docs/install-audit-log-filter.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
+# Install the Audit Log Filter
+
+The `plugin_dir` system variable defines the component library location. If needed, at server startup, set the `plugin_dir` variable.
+
+In the `share` directory, locate the `audit_log_filter_linux_install.sql `script.
+
+At the time you run the script, you can select the database used to store the JSON filter tables. 
+
+* If the component is loaded, the installation script takes the database name from the `audit_log_filter_database` variable
```
Comment (oleksandr-kachan) [inline docs/install-audit-log-filter.md]: audit_log_filter.database

File: docs/install-audit-log-filter.md [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+* If the component is not loaded, but passes the `-D db_name` to the mysql client when the installation script runs, uses the `db_name`.
+* If the component is not loaded and the `-D` option is not provided, the installation script creates the required tables in the default database name `mysql`.
+
+You can also designate a different database with the `audit_log_filter_database` system variable. The database name cannot be NULL or exceed 64 characters. If the database name is invalid, the audit log filter tables are not found.
```
Comment (oleksandr-kachan) [inline docs/install-audit-log-filter.md]: audit_log_filter.database

File: docs/manage-audit-log-filter.md [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+* Consume a large amount of disk space
+* Grow large
+
+You can manage the space by using log file rotation. This operation renames and then rotates the current log file and then uses the original name on a new current log file. You can rotate the file either manually or automatically.
+
+If automatic rotation is enabled, you can prune the log file. This pruning operation can be based on either the log file age or combined log file size.
+
+## Manual log rotation
+
+The default setting for [`audit_log_filter_rotate_on_size`](audit-log-filter-variables.md#audit_log_filter_rotate_on_size) is 1GB. If this option is set to `0`, the audit log filter component does not do an automatic rotation of the log file. You must do the rotation manually with this setting.
```
Comment (oleksandr-kachan) [inline docs/manage-audit-log-filter.md]: audit_log_filter.rotate_on_size

File: docs/manage-audit-log-filter.md [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+
+The `SELECT audit_log_rotate()` command renames the file and creates a new audit log filter file with the original name. You must have the `AUDIT_ADMIN` privilege. 
+
+The files are pruned if either `audit_log_filter_max_size` or `audit_log_filter_prune_seconds` have a value greater than 0 (zero) and `audit_log_filter_rotate_on_size` > 0.
```
Comment (oleksandr-kachan) [inline docs/manage-audit-log-filter.md]: audit_log_filter.max_size, audit_log_filter.prune_seconds, audit_log_filter.rotate_on_size

File: docs/audit-log-filter-overview.md (hunk 1 above)
Comment (alina-derkach-oaza) [reply docs/audit-log-filter-overview.md]: fixed

File: docs/audit-log-filter-overview.md (hunk 2 above)
Comment (alina-derkach-oaza) [reply docs/audit-log-filter-overview.md]: fixed

File: docs/audit-log-filter-overview.md (hunk 3 above)
Comment (alina-derkach-oaza) [reply docs/audit-log-filter-overview.md]: fixed

File: docs/audit-log-filter-variables.md (hunk 4 above)
Comment (alina-derkach-oaza) [reply docs/audit-log-filter-variables.md]: fixed

File: docs/audit-log-filter-variables.md (hunk 5 above)
Comment (alina-derkach-oaza) [reply docs/audit-log-filter-variables.md]: fixed

File: docs/audit-log-filter-variables.md (hunk 6 above)
Comment (alina-derkach-oaza) [reply docs/audit-log-filter-variables.md:341]: updated

File: docs/audit-log-filter-variables.md (hunk 7 above)
Comment (alina-derkach-oaza) [reply docs/audit-log-filter-variables.md]: fixed

File: docs/audit-log-filter-variables.md (hunk 8 above)
Comment (alina-derkach-oaza) [reply docs/audit-log-filter-variables.md]: fixed

File: docs/disable-audit-log-filter.md (hunk 10 above)
Comment (alina-derkach-oaza) [reply docs/disable-audit-log-filter.md]: fixed

File: docs/disable-audit-log-filter.md (hunk 11 above)
Comment (alina-derkach-oaza) [reply docs/disable-audit-log-filter.md]: fixed

File: docs/filter-audit-log-filter-files.md (hunk 12 above)
Comment (alina-derkach-oaza) [reply docs/filter-audit-log-filter-files.md]: fixed

File: docs/install-audit-log-filter.md (hunk 13 above)
Comment (alina-derkach-oaza) [reply docs/install-audit-log-filter.md]: fixed

File: docs/install-audit-log-filter.md (hunk 14 above)
Comment (alina-derkach-oaza) [reply docs/install-audit-log-filter.md]: fixed

File: docs/manage-audit-log-filter.md (hunk 15 above)
Comment (alina-derkach-oaza) [reply docs/manage-audit-log-filter.md]: fixed

File: docs/manage-audit-log-filter.md (hunk 16 above)
Comment (alina-derkach-oaza) [reply docs/manage-audit-log-filter.md]: fixed

File: docs/audit-log-filter-variables.md (hunk 9 above)
Comment (alina-derkach-oaza) [reply docs/audit-log-filter-variables.md]: fixed

File: docs/filter-audit-log-filter-files.md [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+|---|---|---|
+| audit_log_filter_flush() | Manually flush the filter tables | `SELECT audit_log_filter_flush()`
+| audit_log_filter_set_filter() | Defines a filter | `SELECT audit_log_filter_set_filter('log_connections','{ "filter":{}}'`')
+| audit_log_filter_remove_filter() | Removes a filter |
+| audit_log_filter_set_user() | Assigns a filter to a specific user account |
+| audit_log_filter_remove_user() | Removes the filters from a specific user account |
+
+Using a SQL interface, you can define, display, or modify audit log filters. The filters are stored in the `mysql` system database. 
+
+A read-only variable, `audit_log_session_filter_id()`, signals if a filter is assigned to a specific session.
```
Comment (oleksandr-kachan) [inline docs/filter-audit-log-filter-files.md]: "A read-only variable" - Return value of audit_log_session_filter_id() function )

**Review by oleksandr-kachan — CHANGES_REQUESTED** [review]
One more small fix needed

File: docs/filter-audit-log-filter-files.md (hunk 17 above)
Comment (alina-derkach-oaza) [reply docs/filter-audit-log-filter-files.md]: updated

**Review by oleksandr-kachan — APPROVED** [review]

**Pull request merged**